lib/beaker-hostgenerator/cli.rb

```diff
@@ -66,6 +68,14 @@ def initialize(argv = ARGV.dup)
           @options[:list_supported_values] = true
         end
 
+        opts.on('--extract-templates-from-file HOSTS_FILE') do |f|
+          @options[:extract_templates] = f
+        end
+
+        opts.on('--templates-only') do
+          @options[:templates_only] = true
+        end
```

Comment:
I've successfully run facter component pipeline against this branch with the `--templates-only` option. I'm ok leaving this flattened like this for now. I do want to revisit the `=N` issue at a later time, because it complicates the plugin and ci-job-config logic.